openghg/store/base/_datasource.py

```diff
@@ -705,48 +682,49 @@ def key(self) -> str:
         """
         return f"{Datasource._datasource_root}/uuid/{self._uuid}"
 
-    @classmethod
-    def load(
-        cls: Type[T],
-        bucket: str,
-        uuid: str,
-        shallow: bool = False,
-    ) -> T:
-        """Load a Datasource from the object store either by name or UUID
+    def copy_to_memorystore(self, version: str = "latest") -> Dict:
+        """Copy the compressed data for a version from the zarr store into memory.
+
+        Example:
+            memory_store = datasource.copy_to_memorystore(version="v0")
+            with xr.open_zarr(memory_store, consolidated=True) as ds:
+                ...
+
+        This will create a lazily loaded xarray Dataset.
 
-        Args:
-            bucket: Bucket to store object
-            uuid: UID of Datasource
-            key: Key of Datasource
-            shallow: Only load JSON data, do not read Datasets from object store.
-            This will speed up creation of the Datasource object.
         Returns:
-            Datasource: Datasource object created from JSON
+            Dict: In-memory copy of compressed data
         """
-        from openghg.objectstore import get_object_from_json
+        if not self._data_keys:
+            return {}
 
-        key = f"{Datasource._datasource_root}/uuid/{uuid}"
-        data = get_object_from_json(bucket=bucket, key=key)
+        if version == "latest":
+            version = self._latest_version
 
-        return cls.from_data(bucket=bucket, data=data, shallow=shallow)
+        return self._store.copy_to_memorystore(version=version)
 
-    def data(self) -> Dict:
-        """Get the data stored in this Datasource
+    def get_data(self, version: str = "latest") -> xr.Dataset:
+        """Directly accessing a Datasource's data is no longer supported.
```

Comment:
This docstring is confusing: it seems like `get_data` loads the latest version of the data using `xr.open_zarr` (via `LocalZarrStore`)

Reply:
Is the difference between `copy_to_memorystore` and `get_data` that the memory store should be used when modifying data? For computations, it seems like `get_data` is sufficient.

Should `copy_to_memorystore` be more "internal" to the storage parts of openghg, and `get_data` be the interface used by retrieve and analyse?

Reply:
Yeah, I've made `copy_to_memorystore` internal for now and we can see later how we want to use it. I've also updated those docstrings.